pom.xml

```diff
@@ -557,19 +557,19 @@
             <plugin>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <configuration>
-                    <source>${maven.compiler.source}</source>
+                    <source>1.8</source>
```

Comment:
I could not find a better solution after spending some time on this. I am open to alternate solutions if there are any. Without this change, the builds fail with 
`Exit code: 2 - Loading source files for package org.eclipse.collections.codegenerator.model...
error: No source files for package org.eclipse.collections.codegenerator.model`

I tried a few different solutions suggested [here](https://stackoverflow.com/questions/58836862/jdk-11-and-javadoc) and this is the only one that seems to work. I added a new upload artifact step in javadoc workflow to test that it actually generates javadoc with this change.

Reply:
We are using `<version>3.4.1</version>` for maven javadoc plugin. Do we need to upgrade this? It does not smell right that we are linking the 1.8 API docs in Javadoc but building on 11.

Reply:
I started out with 3.5.0, which is the latest version available but rolled back the change when I was running into build errors to eliminate any issues stemming from this plugin version upgrade. 

I can update this to 3.5.0 as it changes nothing for this PR and we will be on the latest version too.

Reply:
I agree it doesn't smell right that we are using 1.8 as the source. One option that I didnt explore is creating an issue [here](https://issues.apache.org/jira/projects/MJAVADOC/summary). I didnt see an issue with our exact problem, so that may be worth creating. I am open to other ideas as well.

Reply:
Created an issue - https://issues.apache.org/jira/browse/MJAVADOC-764